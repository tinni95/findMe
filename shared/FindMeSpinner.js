import { ActivityIndicator, StyleSheet } from "react-native";
import PropTypes from "prop-types";
import React from "react";

const FindMeSpinner = ({ color }) => (
  <ActivityIndicator
    color={color}
    size="large"
    style={styles.activityIndicator}
  />
);

FindMeSpinner.propTypes = {
  color: PropTypes.string
};

FindMeSpinner.defaultProps = {
  color: "blue"
};

const styles = StyleSheet.create({
  activityIndicator: {
    flex: 1,
    justifyContent: "flex-start"
  }
});

export default FindMeSpinner;
