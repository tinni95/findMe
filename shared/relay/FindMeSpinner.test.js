import React from 'react';
import { render } from 'react-native-testing-library';

import FindMeSpinner from './FindMeSpinner';

describe('FindMeSpinner', () => {
  describe('with no non-default props', () => {
    it('renders successfully to match snapshot', () => {
      const rendered = render(<FindMeSpinner />);
      expect(rendered.toJSON()).toBeTruthy();
      expect(rendered.toJSON()).toMatchSnapshot();
    });
  });

  describe('with a different color prop', () => {
    it('renders successfully to match snapshot', () => {
      const rendered = render(<FindMeSpinner color="#ddd" />);
      expect(rendered.toJSON()).toBeTruthy();
      expect(rendered.toJSON()).toMatchSnapshot();
    });
  });
});
