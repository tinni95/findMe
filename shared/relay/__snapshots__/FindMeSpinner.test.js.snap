// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FindMeSpinner with a different color prop renders successfully to match snapshot 1`] = `
<ActivityIndicator
  animating={true}
  color="#ddd"
  hidesWhenStopped={true}
  size="large"
  style={
    Object {
      "flex": 1,
      "justifyContent": "flex-start",
    }
  }
/>
`;

exports[`FindMeSpinner with no non-default props renders successfully to match snapshot 1`] = `
<ActivityIndicator
  animating={true}
  color="blue"
  hidesWhenStopped={true}
  size="large"
  style={
    Object {
      "flex": 1,
      "justifyContent": "flex-start",
    }
  }
/>
`;
